controllers/datadogagent/override/podtemplatespec.go

```diff
@@ -119,7 +190,6 @@ func PodTemplateSpec(manager feature.PodTemplateManagers, override *v2alpha1.Dat
 
 func overrideCustomConfigs(manager feature.PodTemplateManagers, customConfs map[v2alpha1.AgentConfigFileName]v2alpha1.CustomConfig, componentName v2alpha1.ComponentName, ddaName string) {
 	for _, customConfig := range customConfs {
-		// Note: customConfig.ConfigData requires creating a configmap, so it cannot be handled here
```

Comment:
the configMap v. configData volumes will be handled properly in the same subsequent PR